contracts/DefaultStrategy.sol

```diff
@@ -0,0 +1,497 @@
+     * Election.sol.
+     */
+    error NoVotableGroups();
+
+    /**
+     * @notice Emitted when a deprecated group is no longer being voted for and
+     * the contract forgets about it entirely.
+     * @param group The group's address.
+     */
+    event GroupRemoved(address indexed group);
+
+    /**
+     * @notice Used when attempting to activate a group that is already active.
+     * @param group The group's address.
+     */
+    error GroupAlreadyAdded(address group);
+
+    /**
+     * @notice Used when an attempt to remove a deprecated group from the
+     * EnumerableSet fails.
+     * @param group The group's address.
+     */
+    error FailedToRemoveDeprecatedGroup(address group);
+
+    /**
+     * @notice Used when a group does not meet the validator group health requirements.
+     * @param group The group's address.
+     */
+    error GroupNotEligible(address group);
+
+    /**
+     * @notice Used when attempting to activate a group when the maximum number
+     * of groups voted (as allowed by the Election contract) is already being
+     * voted for.
+     */
+    error MaxGroupsVotedForReached();
+
+    /**
+     * @notice Initialize the contract with registry and owner.
+     * @param _registry The address of the Celo registry.
+     * @param _owner The address of the contract owner.
+     * @param _manager The address of the Manager contract.
+     */
+    function initialize(
+        address _registry,
+        address _owner,
+        address _manager
+    ) external initializer {
+        _transferOwnership(_owner);
+        __UsingRegistry_init(_registry);
+        __Managed_init(_manager);
+    }
+
+    /**
+     * @notice Set this contract's dependencies in the StakedCelo system.
+     * @param _account The address of the Account contract.
+     * @param _groupHealth The address of the Group health contract.
+     * @param _allowedStrategy The address of the AllowedStrategy contract.
+     */
+    function setDependencies(
+        address _account,
+        address _groupHealth,
+        address _allowedStrategy
+    ) external onlyOwner {
+        require(_account != address(0), "account null address");
+        require(_groupHealth != address(0), "validate group null address");
+        require(_allowedStrategy != address(0), "allowedStrategy null address");
+
+        groupHealthContract = IGroupHealth(_groupHealth);
+        allowedStrategyContract = IAllowedStrategy(_allowedStrategy);
+        account = IAccount(_account);
+    }
+
+    /**
+     * @notice Sorts an array of GroupWithVotes structs based on increasing
+     * `votes` values.
+     * @param groupsWithVotes The array to sort.
+     * @dev This is an in-place insertion sort. In general in Solidity we should
+     * be careful of algorithms on arrays, especially O(n^2) ones, but here
+     * we're guaranteed to be working with a small array, its length is bounded
+     * by the maximum number of groups that can be voted for in Elections.sol.
```

Comment:
I guess this is not true if the limit is lifted. Should we have an internal limit in this contract for the number of activated groups?
RE: my comment on the `MaxGroupsVotedForReached` event, could then still be useful, but update the comment to refer to the internal limit rather than the Election one.

Reply:
We should probably discuss this separately - I already mentioned this to Mathew and we will be having a meeting (also with you) where we will be discussing how to update default strategy